pkg/cmd/cli/cmd/set/env.go

```diff
@@ -262,6 +262,9 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	if err != nil {
 		return err
 	}
+	if len(env) == 0 {
```

Comment:
Hm, it does, I could add a check to make sure `--list` is not set before returning this error